v1.1/create-security-certificates-openssl.md

```diff
@@ -0,0 +1,292 @@
+-----------|------
+[`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) | Generate an RSA private key
+[`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) | Create a private key and a CSR (certificate signing request)
+[`openssl ca`](https://wiki.openssl.org/index.php/Manual:Ca(1)) | Issue a root CA certificate based on the CSR.
+
+## Certificate Directory
+
+Use the `openssl genrsa` and `openssl req` subcommands to generate the CA certificate and all node and client certificates and keys in a single directory, with the files named as follows:
+
+File name pattern | File usage
+-------------|------------
+`ca.crt`     | CA certificate
+`node.crt`   | Server certificate
+`node.key`   | Key for server certificate
+`client.<user>.crt` | Client certificate for `<user>` (eg: `client.root.crt` for user `root`)
+`client.<user>.key` | Key for the client certificate
+
+Note the following:
+
+- The CA key is never loaded automatically by `cockroach` commands, so it should be created in a separate directory, identified by the `--ca-key` flag.
+
+- <span style="color:red">Is this applicable to OpenSSL?</span> Keys (files ending in `.key`) must not have group or world permissions (maximum permissions are 0700, or `rwx------`). This check can be disabled by setting the environment variable `COCKROACH_SKIP_KEY_PERMISSION_CHECK=true`.
+
+## Configuration Files
+
+To use `openssl req` and `openssl ca` subcommands, you need the following configuration files:
+
+File name pattern | File usage
+-------------|------------
+`ca.conf`     | CA configuration file
+`node.conf`   | Server configuration file
+`client.conf` | Client configuration file
+
+## Examples
+
+### Generate the CA key and certificate pair
+
+1. Create two directories:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir certs
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir my-safe-directory
+    ~~~
+    - `certs`: You'll generate your CA certificate and all node and client certificates and keys in this directory and then upload some of the files to your nodes.
+    - `my-safe-directory`: You'll generate your CA key in this directory and then reference the key when generating node and client certificates. After that, you'll keep the key safe and secret; you will not upload it to your nodes.
+
+2. Create the ca.conf file
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    [ ca ]
+    default_ca = CA_default
+
+    [ CA_default ]
+    default_days = 3660
+    database = index.txt
+    serial = serial.txt 
+    default_md = sha256 
+    copy_extensions = copy
+
+    # Used to generate the CA certificate.
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    x509_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = Cockroach CA
+
+    [ extensions ]
+    keyUsage = critical,digitalSignature,nonRepudiation,keyEncipherment,keyCertSign
```

Comment:
We should point out the important part of the file.
For `ca.conf`, that's `keyUsage` and `extendedKeyUsage` in multiple sections.

Reply:
"Do not change this, else CRDB won't work".

Reply:
Done.